src/device/pf_cmsm.c

```diff
@@ -183,38 +183,40 @@ int pf_cmsm_rm_read_ind(
 {
    int                     ret = -1;
 
-   switch (p_ar->cmsm_state)
+   if (p_ar != NULL)
```

Comment:
p_ar != NULL is checked twice in some functions (the second time is always false). Also, in some functions p_ar != NULL is checked after deferencing p_ar. 

Make all functions similar with regards to p_ar if possible, and vaildate parameters and abort early, rather than adding extra indentation levels.

Reply:
Done